src/Spatial/Euclidean/Line2D.cs

```diff
@@ -294,5 +328,26 @@ void IXmlSerializable.WriteXml(XmlWriter writer)
             writer.WriteElement("StartPoint", StartPoint);
             writer.WriteElement("EndPoint", EndPoint);
         }
+
+        /// <summary>
+        /// Compute the signed distance from this `line` to the given point `p`. If the sign of the distance is positive, the normal vector from the line is in the same side to the point. 
+        /// </summary>
+        /// <param name="p"></param>
+        /// <returns></returns>
+        public double SignedDistanceTo(Point2D p)
```

Comment:
Using the notation you employed above for the intersection case, the line is already represented as `q=s+t*d`, where `s` is the start point and `d` is the (unit) direction vector (both of these are cartesian 2D vectors).

Then the distance of any point `p` to this line `q` can be expressed as the norm of this vector `s - (p-s)*d`

Reply:
Thanks for the comment.

If I understand your comment correctly, it seems to me that `s - (p-s)*d` does not mean the vector from the point `p` to the point on the line `q`. 
Because `(p-s)` is the vector.
The operator `*` is for the 2 vectors `(p-s)` and `d`, I cannot understand in this context.

Should I use the normal vector of the line `q`?

Reply:
Actually my original comment war not correct, the correct one should be `s - [(p-s)*d] * d`.
`p - s` is a vector, `(p-s)*d` is a dot product of two vectors (a real number, and then `[(p-s)*d] * d` is again a vector

Reply:
> The more I think about it, the less I think that the notion of "signed" distance of point from a line is something meaningful.

True. You can say that again.

> I think we should provide only one method for the distance, which returns either 0 (point on the line) or a positive value (point not on the line).

From now on, I will call the method `AbsoluteDistance`.
If you don't mind, I will try to implement `AbsoluteDistance` method.
Because, if there are few user stories to use `Line2D.SignedDistance(point)`, 
implementing just `AbsoluteDistance` looks make sense. 
If you mind, please let me know to change my mind.


> I am not sure what you mean exactly by "tangential component", I would call this "the normal component w.r.t. the line"(snip)

Sorry for making you confused by the figure violating the naming convention.
I should have shown you the figure showing the interface between 2 regions like this:
(The figure bellow was taken from https://engineering.purdue.edu/wcchew/ece604f19/EMFTAll20191204.pdf p.31 4.3.1 Faraday's law)

![image](https://github.com/mathnet/mathnet-spatial/assets/1047493/5e376d93-2ae3-42ec-8203-48775a795caa)

For this issue, I arranged this figure imaginally so that the interface is a straight line.

Reply:
> From now on, I will call the method `AbsoluteDistance`. If you don't mind, I will try to implement `AbsoluteDistance` method. Because, if there are few user stories to use `Line2D.SignedDistance(point)`, implementing just `AbsoluteDistance` looks make sense. If you mind, please let me know to change my mind.

I think `line.DistanceFrom(p)` of `line.DistanceTo(p)` is straight forward. The `Absolute` does not any value to the function itself, I would argue that it adds confusion.

> Sorry for making you confused by the figure violating the naming convention. I should have shown you the figure showing the interface between 2 regions like this

Stil though, the distance of the line from the point is along the normal direction `n`, and not along the tangential component `t`.

Reply:
> I think line.DistanceFrom(p) of line.DistanceTo(p) is straight forward. The Absolute does not add any value to the function itself, I would argue that it adds confusion.

Thank you for arguing this.
Calling `AbsoluteDistance` help us go ahead with this issue.
Your opinion is reasonable.
Is `line.DistanceTo(p)` fine for you?
Either is fine, but this one for now.



> > Sorry for making you confused by the figure violating the naming convention. I should have shown you the figure showing the interface between 2 regions like this
> 
> Stil though, the distance of the line from the point is along the normal direction `n`, and not along the tangential component `t`.

You have a point.
That is what I wanted to say.

Your original comment was:

> Then the distance of any point p to this line q can be expressed as the norm of this vector `s - (p-s)*d`

Your revised comment was:

> Actually my original comment war not correct, the correct one should be `s - [(p-s)*d] * d`.

In my understanding, `s - [(p-s)*d] * d` is along the tangental component `t` (as I quoted the top part of this comment), and I think it is different from the vector which we wanted when we calculate the distance from the line to the given point `p`. We need another expression other than ones you have provided.